pyproject.toml

```diff
@@ -1,164 +1,123 @@
-    "License :: OSI Approved :: Apache Software License",
-    "Natural Language :: English",
-    "Operating System :: OS Independent",
-    "Programming Language :: Python :: 3",
-    "Programming Language :: Python :: 3 :: Only",
-    "Topic :: Scientific/Engineering :: Artificial Intelligence",
-    "Topic :: Scientific/Engineering :: Image Processing",
-    "Topic :: Software Development :: Libraries",
-]
-dependencies = [
-    "packaging",
-    "torch>=1.9.1",
-]
-[project.optional-dependencies]
-dev = [
-    "isort",
-    "kornia-rs==0.0.8",
-    "mypy[reports]",
-    "numpy",
-    "opencv-python",
-    "pre-commit>=2",
-    "pydocstyle",
-    "pytest==7.2.1",
-    "pytest-cov==4",
-    "scipy",
+  "Development Status :: 4 - Beta",
+  "Environment :: Console",
+  "Environment :: GPU",
+  "Intended Audience :: Developers",
+  "Intended Audience :: Education",
+  "Intended Audience :: Information Technology",
+  "Intended Audience :: Science/Research",
+  "License :: OSI Approved :: Apache Software License",
+  "Natural Language :: English",
+  "Operating System :: OS Independent",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3 :: Only",
+  "Topic :: Scientific/Engineering :: Artificial Intelligence",
+  "Topic :: Scientific/Engineering :: Image Processing",
+  "Topic :: Software Development :: Libraries",
 ]
-docs = [
-    "furo",
-    "matplotlib",
-    "opencv-python",
-    "PyYAML<6.1.0,>=5.1",
-    "sphinx>=4",
-    "sphinx-autodoc-defaultargs",
-    "sphinx-autodoc-typehints",
-    "sphinx-copybutton>=0.3",
-    "sphinx-design",
-    "sphinx-rtd-theme>0.5",
-    "sphinxcontrib-bibtex",
-    "sphinxcontrib-gtagjs",
-    "sphinxcontrib-youtube",
-    "torchvision",
-]
-x = [
-    "accelerate==0.16",
+dynamic = [
+  "dependencies",
+  "optional-dependencies",
+  "readme",
+  "version",
 ]
 [project.urls]
-"Bug Tracker" = "https://github.com/kornia/kornia/issues"
-Documentation = "https://kornia.readthedocs.io/en/latest"
-Download = "https://github.com/kornia/kornia"
-Homepage = "https://www.kornia.org"
-"Source Code" = "https://github.com/kornia/kornia"
+  "Bug Tracker" = "https://github.com/kornia/kornia/issues"
+  Documentation = "https://kornia.readthedocs.io/en/latest"
+  Download = "https://github.com/kornia/kornia"
+  Homepage = "https://www.kornia.org"
+  "Source Code" = "https://github.com/kornia/kornia"
+
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [
+  "setuptools>=61.2",
+]
 
 [tool.setuptools]
-zip-safe = true
-license-files = ["LICENSE"]
-include-package-data = false
+  zip-safe = true
+  license-files = ["LICENSE"]
+  include-package-data = false
+
+[tool.setuptools.dynamic]
+  dependencies = {file = ["requirements/requirements.txt"]}
+  version = {attr = "kornia.__version__"}
+  readme = {file = ["README.md"], content-type = "text/markdown"}
+
+  [tool.setuptools.dynamic.optional-dependencies]
+    dev = {file = "requirements/requirements-dev.txt"}
```

Comment:
Why in file ? Btw, seems that after the previous pr around this fixed the issue with the dependency graph visibility

Reply:
for me is showing this ![](https://user-images.githubusercontent.com/20444345/221412323-614853d3-8af4-480e-b78e-d652f74a3459.png)

And requirements files seem to work fine -- before when we has `setup.py`, isn't fully working too, if I'm not wrong, he just detected the package requirements, but no the optional like pytest